### ISSUE TITLE
Make returnify error more readable for models where the last statement is not an expression

### ISSUE THREAD

**Issue by stuhlmueller** [probmods/webppl#489]
This is a subset of #331

**Comment by chrisranderson** [comment]
I'm guessing this is more than changing the string on this line: https://github.com/probmods/webppl/blob/dev/src/syntax.js#L117. If not, I can take a whack at it. :)

**Comment by null-a** [comment]
> I'm guessing this is more than changing the string on this line:

Perhaps, but only a *little* more...

The problematic error arises when trying to run a program like this:

```js
var a = 0;
```

Which produces:

```
{ type: 'VariableDeclaration',
  declarations: 
   [ { type: 'VariableDeclarator',
       id: [Object],
       init: [Object],
       range: [Object],
       loc: [Object] } ],
  kind: 'var',
  range: [ 0, 9 ],
  loc: 
   { start: { line: 1, column: 0 },
     end: { line: 1, column: 9 },
     source: 'test.wppl' } }
returnify
path/to/webppl/webppl:56
      throw error;
      ^

Error: returnify
    at path/to/webppl/src/syntax.js:20:11
    ... more stack trace ...
```

That error is generated by the function returned from `fail` on the line you mentioned.

I imagine that making following changes would probably produce acceptable results:

1. Don't show the ast node.
2. Have a more descriptive message, explain how to fix the problem.
3. Perhaps loose the JS stack trace which doesn't seem useful for users. We could `throw` a string rather than an `Error` to achieve this.

So, perhaps not *too* much more than just changing the string?

**Comment by chrisranderson** [comment]
How about this: if the user doesn't return something, we return the empty string for them. Maybe they just want to check syntax and don't care to see what the program returns, or they print the result themselves. Maybe they had a return value, but didn't want it to print out so they commented it out. Instead of changing the error message, we just drop the requirement that the program have a return value. Thoughts? I think that beats `Error: programs must end with a return value.`

**Comment by null-a** [comment]
Yeah, this crossed my mind, and I think it's worth considering. If there are reasons why it wouldn't work out I don't know what they are.

This *might* be a bigger change than just changing the error, but I'm not sure what's involved without looking closely. If we're end up short of time, just changing the error would be still be a big improvement and worth having for v0.9.

> we return the empty string for them

I think I would use `undefined` instead, and then consider separately the question of whether to suppress output when the program evaluates to `undefined`. One reason is that functions that end with a statement rather than an expression already return `undefined`.

**Comment by null-a** [comment]
Closed by #587.
